### DISCUSSION TITLE
Exon skipping / alternative splice sites as inconsistencies instead of new isoforms

### DISCUSSION THREAD

**Discussion by akurlovs** [ablab/IsoQuant#115] [General]
Hello! What I have noticed when running this tool (and after trying various settings) is that it seems to be designed to label exon skipping and alternative splice sites as "inconsistencies" instead of reporting them as isoforms in the updated GTF. So if I am interested in (for instance) how often different exons of a particular gene get skipped (together and/or individually), and this exon skipping is not in the original GTF provided when running IsoQuant, the only way to get this information from the output is by parsing one of the read files (the exon count files only provides this info on each putative exon individually, not whether e.g., exon 1 and exon 3 were skipped by the same reads).

Am I missing a way to circumvent this? Why are these events labeled as inconsistencies instead of new isoforms in the updated GTF when exon skipping is the most common form of alternative splicing?

**Comment by andrewprzh** [comment]
Dear @akurlovs 

Sorry for the delay - I was out of the office for a while.

IsoQuant basically has two independent stages. First one is the reference-bases analysis, which involves read-to-isoform assignment and classification, and gene/isoform/exon quantification. At this stage no novel isoforms are predicted, and thus reads representing novel isoforms (including novel isoforms with skipped exons) are classified as inconsistent with the annotation. This information is stored in `.read_assignments.tsv`.

During the second stage, IsoQuant predicts novel isoforms. If there are enough reads to support a novel isoform (including ones with skipped exons), this novel isoform will be reported in `.transcript_models.gtf`. Novel isoforms are marked accordingly (with nic or nnic in their id). The reported transcript models are also quantified and the counts are stored in `.transcript_model_counts.tsv`. 

Finally, to figure out which novel isoforms have skipped exons with respect to a reference isoform, you may use `--sqanti_output` option, which will provide a table showing relation between novel and reference isoforms.

In summary, IsoQuant does report transcript model with skipped exons during its second stage. Hope that helps, don't hesitate to ask more questions.

Best
Andrey
